website/blog/2025-02-12-best-alternative-to-mongodb-data-api-and-http-endpoints.md

```diff
@@ -0,0 +1,199 @@
+---
+slug: best-alternative-to-mongodb-data-api-and-http-endpoints
+title: 'The Best Alternative for MongoDB Data API and HTTP Endpoints'
+authors: [alex]
+description: >
+  MongoDB's deprecation of Atlas Data API left many developers without a suitable replacement. FerretDB v2 now provides a compatible alternative for the deprecated Atlas Data API.
+image: /img/blog/mongodb-data-api.jpg
+tags: [open source, sspl, product, document databases, community]
+---
+
+![The Best Alternative for MongoDB Data API and HTTP Endpoints](/img/blog/mongodb-data-api.jpg)
+
+In September 2024, MongoDB announced the deprecation of Atlas Data API – among other features – leaving many developers affected, and without a suitable replacement.
+We are happy to announce that [FerretDB v2](https://blog.ferretdb.io/ferretdb-releases-v2-faster-more-compatible-mongodb-alternative) now provides a compatible replacement for the deprecated Atlas Data API.
+
+<!--truncate-->
+
+The Data API deprecation affected many developments, especially as it came without alternative migration options.
+This is a common theme for many proprietary (or license-restrictive) projects, and not just MongoDB – [read more on the uncertainties of proprietary solutions here](https://blog.ferretdb.io/why-open-source-important-proprietary-uncertainties).
+Open source solutions like FerretDB offer control, community support, and long-term stability, so you're never left stranded.
+
+The good thing is: with the release of FerretDB v2, you can successfully replace MongoDB Data API and perform database operations on FerretDB using a direct HTTP-based method to access and interact with their databases.
+
+In this post, we'll walk you through how to use FerretDB's Data API, demonstrating how you can find, insert, update, and delete documents – all without needing a MongoDB server.
+
+## Why Data API matters
+
+Some programming environments just don't have a direct way to connect to a database – no native MongoDB driver, no built-in support, nothing.
```

Comment:
> Some programming environments just don't have a direct way to connect to a database

In our case we need a direct access to database, as the api is a part of FerretDB process, and it's not a standalone service.

> no native MongoDB driver, no built-in support, nothing

What do we mean by built-in support? I guess it could be shorten to just mentioning that it can be used in cases where driver is not available or cannot be used.

Reply:
Thanks for the review. I've updated it and also added that bit to the next section.